Modules/Loadable/Markups/VTKWidgets/vtkSlicerMarkupsWidgetRepresentation3D.cxx

```diff
@@ -633,6 +651,23 @@ void vtkSlicerMarkupsWidgetRepresentation3D::UpdateFromMRML(vtkMRMLNode* caller,
     controlPoints->TextProperty->SetOpacity(this->MarkupsDisplayNode->GetOpacity() * hierarchyOpacity);
     controlPoints->TextProperty->SetFontSize(static_cast<int>(5. * this->MarkupsDisplayNode->GetTextScale()));
 
+    controlPoints->OccludedProperty->DeepCopy(controlPoints->Property);
+    if (this->MarkupsDisplayNode->GetOccludedVisibility() && this->MarkupsDisplayNode->GetOccludedOpacity() > 0.0)
+      {
+      // If OccludedVisibility is enabled, then the depth of the widget will be 0.0.
+      // The point should never be occluded, so set tolerance to 1.0.
+      controlPoints->SelectVisiblePoints->SetTolerance(1.0);
```

Comment:
It would be nice if we would still use point visibility and display occluded labels with OccludedOpacity. If this is complicated then just write there in a comment that it could be done and what are the complications.

Reply:
It's possible if we change when the vtkSelectVisiblePoints filter is updated, as it would need to be updated before the points are rendered.
I don't think it should be too complicated.

Reply:
Should occluded control points be interactive?

Reply:
I cannot think of an application that would require the user to move always-on-top markups. Because it would be hard to intuitively feel which way they go in 3D. But I can imagine that people would like to click on them to center the 2D views on a particular fiducial, or show other annotations. E.g. imagine an interactive anatomical atlas. People could click on these points inside the 3D anatomy to e.g. display their names, or some measurements related to them.

Reply:
I agree, it would be very useful to be able to interact with all visible landmarks, even if occluded. Maybe in some cases it could be useful to lock them, but I don't have anything specific application in mind, so keeping them interactive for now should be good.